oak_channel/SPEC.md

```diff
@@ -44,91 +43,47 @@ The service MUST NOT send request messages.
 ##### Handling an invocation
 
 The service MUST respond to each request message with a single response message.
-It SHOULD respond to request messages in first-in-first-out order, but MAY
-prioritize some request messages. The service MAY send a response message before
-the request message has been received in its entirety. Sending a response
-message concludes the invocation.
+The service MAY send a response message before the request message has been
+received in its entirety, but MUST NOT send start sending a response before any
+part of the request message has been received. Sending a response message
```

Comment:
I'm not sure if the response should be called a message, since this is the invocation layer and not the message layer.
I think it's better to call it "Invocation request" and "Invocation response". And keep the "message" only for describing the "Message layer" entities.